packages/esm-patient-registration-app/src/patient-registration/field/name/name-field.component.tsx

```diff
@@ -68,13 +69,17 @@ export const NameField = () => {
         )}
 
         <div className={styles.nameField}>
-          <div className={styles.dobContentSwitcherLabel}>
-            <span className={styles.label01}>{t('patientNameKnown', "Patient's Name is Known?")}</span>
-          </div>
-          <ContentSwitcher className={styles.contentSwitcher} onChange={toggleNameKnown}>
-            <Switch name="known" text={t('yes', 'Yes')} />
-            <Switch name="unknown" text={t('no', 'No')} />
-          </ContentSwitcher>
+          {allowUnknownName && (
+            <>
```

Comment:
What about the edge case where "nameKnown" = false?  For example, maybe for a while the system is configured to allow uknown patients, and then the implementation wants to remove this feature?  In this case, we'd likely want to show the "Unknown/Known" section for cases where you are editing a patient that is already configured as "unknown", even if normally this is not displayed.  Maybe have this be (allowUnknownName || !nameKnown)?

Reply:
Hey @mseaton. Taking a deep thought into this, I think this is sufficient. I created a patient with an Unknown name and an estimated date of birth then tried editing the patient details. See the screencast below. 

The constant `nameKnown` is responsible for toggling **yes** or **no**, that is when the `unidentifiedPatient` config is enabled (true). It is not part of the configuration.
[Screencast from 2023-05-15 18-59-39.webm](https://github.com/openmrs/openmrs-esm-patient-management/assets/104269786/527c3edc-63fd-417b-ac55-f3c797c1dbd5)

Reply:
So briefly, if a patient was created as unknown before, we should show the toggle while editing.

Reply:
Yes, correct.

Reply:
Hi @mseaton!
I am working on taking this PR forward.

I have a query. According to the above scenario:
An unknown patient is registered. Then the implementation disables allowing the unknown patient. What should happen in this case if the above patient's details are open to edit. In this case, which case should we follow more precisely?

1. We show the toggle for the `Patient is Known` toggle? In this case 2 scenarios arise:

1.a. If the user (nurse/ provider) **doesn't** toggle the Patient is Known -> Yes, should we allow saving the updates to that patient?
1.b. If the user toggles to Patient is known -> Yes, should we hide the toggle so that the user cannot set the toggle to No again, or should we allow the same?

2. We don't show the "Patient is Known" toggle and automatically set the names to empty input boxes and set the attribute value of "Unknown patient" to false ?

I will be handling the "Unknown patient" attribute in both the cases, but which user scenario to move ahead with, please let me know.
Thanks!

Reply:
CC: @donaldkibet @rbuisson @ibacher

Reply:
@vasharma05 - if the patient being edited has a person attribute indicating that they are an unknown patient, then when we edit this patient we need to show this.  Both for completeness and accuracy of viewing the data, but also to ensure that this can be edited if desired.  Otherwise, for any patient that might have been marked as unknown historically (or maybe imported or migrated in this way) will never be able to have this attribute changed.

The same is true for the estimated birthdate field, which came up in this thread as a parallel issue.

So we just need to toggle this section as visible if a) it is configured as visible in the app OR b) if the patient is already an unknown patient based on their existing data.

Reply:
I agree with @mseaton on this.

Basically, as I understand it, the ask here is that we display the toggle:

1. If configured to do so
2. If not configured to do so, but editing a patient with an unknown name / estimated birthdate

Which seems reasonable.

A further note on this: the patient attribute is intended to be the "source of truth" for whether the patient's name is known or not. The string "UNKNOWN UNKNOWN" is a default. We actually let the implementer configure this value. The intention is to support i18n, since even patient's with an unknown name have whatever we store in the name displayed in a few places. For instance, for an implementation primarily used in a French-speaking country, "INCONNU INCONNU" is probably preferable.

Similarly, with birthdate, we have a `birthdate_estimated` column in the backend which should be used to determine whether a patient's birthdate is estimated. If you're using the REST API, this is just another boolean variable returned with the patient; if you're using the FHIR API, the birthdate will not have a day, but just a year or year and month).